karapace/protobuf/schema.py

```diff
@@ -106,6 +109,49 @@ def option_element_string(option: OptionElement) -> str:
     return f"option {result};\n"
 
 
+@dataclasses.dataclass
+class UsedType:
+    parent_object_qualified_name: str
+    used_attribute_type: str
+
+
+@dataclasses.dataclass(frozen=True)
+class TypeTree:
+    token: str
+    children: List["TypeTree"]
+
+    @property
+    def is_full_path_type(self) -> bool:
```

Comment:
I am not sure of the name, just a proposal below. I also did consider here `is_leaf` or `is_terminal_node` as this is a tree structure. But domain related name here is a lot better.
```suggestion
    def is_fully_qualified_type(self) -> bool:
```